src/Appium.Net/Appium/Android/AndroidDriver.cs

```diff
@@ -287,21 +287,67 @@ public IList<string> GetPerformanceDataTypes() =>
 
         #region Device Interactions
 
-        /**
-        * This method locks a device.
-        */
-        public void Lock() => AppiumCommandExecutionHelper.Lock(this, 0);
+        /// <summary>
+        /// Locks the device. Optionally, unlocks it after a specified number of seconds.
+        /// </summary>
+        /// <param name="seconds">
+        /// The number of seconds after which the device will be automatically unlocked. 
+        /// Set to 0 or leave it empty to require manual unlock.
+        /// </param>
+        /// <exception cref="WebDriverException">Thrown if the command execution fails.</exception>
+        public void Lock(int? seconds = null)
+        {
+            var parameters = new Dictionary<string, object>();
+
+            if (seconds.HasValue && seconds.Value > 0)
+            {
+                parameters["seconds"] = seconds.Value;
+            }
+
+            ExecuteScript("mobile: lock", parameters);
```

Comment:
You could add try/except and fallback to the legacy one like https://github.com/appium/python-client/blob/e4f06abf1f6bc302bde373071d1e3c007a67c03a/appium/webdriver/extensions/hw_actions.py#L40-L47 does. Then, maybe the change doesn't need to be a "breaking" one since the existing one will do the same

Reply:
The "breaking" is more for the `unlock` method that now requires additional parameters

Reply:
I also see there is a a todo on that fallback
https://github.com/appium/python-client/blob/e4f06abf1f6bc302bde373071d1e3c007a67c03a/appium%2Fwebdriver%2Fextensions%2Fhw_actions.py#L45
So isn't it redundant?

Reply:
Yea, the todo is correct. We haven't decided when we completely remove them, though. I'm good either, just I brought up the idea here